### ISSUE TITLE
Chef Infra msi download doesn't work anymore (cannot bootstrap a windows node)

### ISSUE THREAD

**Issue by TiTi** [chef/chef#11394]
## Description
We are not able to bootstrap windows nodes anymore.
The .msi downloaded is not a valid installation package, but an HTML response.
Error in logs:
`WARNING: Failed to install Chef Infra Client MSI package in remote context with status code 1620.`

knife bootstrap tries to download from:
https://www.chef.io/chef/download?p=windows&pv=2016&m=x86_64&DownloadContext=PowerShell&channel=stable&v=15.14.0
See source here: https://github.com/chef/chef/blob/master/knife/lib/chef/knife/core/windows_bootstrap_context.rb#L299 (no recent change)
But this doesn't work anymore (whatever the version). We get a 52KB HTML response rather than a ~180Mo .msi:
![image](https://user-images.githubusercontent.com/137691/115243311-15680d00-a123-11eb-9c9c-b27b7fefa5e0.png)

## Chef Version
15.14.0
but same url issue with 16.13.16

## Platform Version
windows server 2016

## Replication Case
Bootstrap a windows server 2016 machine using `knife bootstrap`

## Client Output
(Sensitive data as been replaced)
```
knife bootstrap 10.98.12.4 --node-name <nodename> --connection-protocol winrm --connection-user <user> --connection-password *** --bootstrap-vault-json {\"passwords\": [\"pass1\", \"pass2\"], \"accesskeys\": [\"key1\"]} --policy-name mypolicy --policy-group tower_x --bootstrap-version 15.14.0 --yes
```

(Some chars might come from our CI/CD tools)
```
...

No existing installation of Chef Infra Client detected
Checking for existing downloaded package at \"C:\\Users\\<user>\\AppData\\Local\\Temp\\chef-client-latest.msi\"
Found existing downloaded package, deleting.
Attempting to download client package using PowerShell if available...
powershell.exe -ExecutionPolicy Unrestricted -InputFormat None -NoProfile -NonInteractive -File  C:\\chef\\wget.ps1 \"https://www.chef.io/chef/download?p=windows&pv=2016&m=x86_64&DownloadContext=PowerShell&channel=stable&v=15.14.0\" \"C:\\Users\\<user>\\AppData\\Local\\Temp\\chef-client-latest.msi\"
Download via PowerShell succeeded.
Installing downloaded client package...

C:\\Users\\<user>\\Documents>msiexec /qn /log \"C:\\Users\\<user>\\AppData\\Local\\Temp\\chef-client-msi6220.log\" /i \"C:\\Users\\<user>\\AppData\\Local\\Temp\\chef-client-latest.msi\" 
T\u0000h\u0000i\u0000s\u0000 \u0000i\u0000n\u0000s\u0000t\u0000a\u0000l\u0000l\u0000a\u0000t\u0000i\u0000o\u0000n\u0000 \u0000p\u0000a\u0000c\u0000k\u0000a\u0000g\u0000e\u0000 \u0000c\u0000o\u0000u\u0000l\u0000d\u0000 \u0000n\u0000o\u0000t\u0000 \u0000b\u0000e\u0000 \u0000o\u0000p\u0000e\u0000n\u0000e\u0000d\u0000.\u0000 \u0000 \u0000C\u0000o\u0000n\u0000t\u0000a\u0000c\u0000t\u0000 \u0000t\u0000h\u0000e\u0000 \u0000a\u0000p\u0000p\u0000l\u0000i\u0000c\u0000a\u0000t\u0000i\u0000o\u0000n\u0000 \u0000v\u0000e\u0000n\u0000d\u0000o\u0000r\u0000 \u0000t\u0000o\u0000 \u0000v\u0000e\u0000r\u0000i\u0000f\u0000y\u0000 \u0000t\u0000h\u0000a\u0000t\u0000 \u0000t\u0000h\u0000i\u0000s\u0000 \u0000i\u0000s\u0000 \u0000a\u0000 \u0000v\u0000a\u0000l\u0000i\u0000d\u0000 \u0000W\u0000i\u0000n\u0000d\u0000o\u0000w\u0000s\u0000 \u0000I\u0000n\u0000s\u0000t\u0000a\u0000l\u0000l\u0000e\u0000r\u0000 \u0000p\u0000a\u0000c\u0000k\u0000a\u0000g\u0000e\u0000.\u0000
\u0000
\u0000WARNING: Failed to install Chef Infra Client MSI package in remote context with status code 1620.
WARNING: This may be due to a defect in operating system update KB2918614: http://support.microsoft.com/kb/2918614
WARNING: Saving installation log of failure at \"C:\\Users\\<user>\\AppData\\Local\\Temp\\chef-client-msi6220.log-fail.log\"
WARNING: Retrying installation with local context...
SUCCESS: The scheduled task \"chefclientbootstraptask\" has successfully been created.
Successfully created scheduled task to install Chef Infra Client.
SUCCESS: Attempted to run the scheduled task \"chefclientbootstraptask\".
Successfully started Chef Infra Client installation scheduled task.
Waiting for installation to complete -- this may take a few minutes...

SUCCESS: Signal received.
Finished waiting for Chef Infra Client package to install.
Installation completed successfully
Writing validation key...
Validation key written.
```

## Notes
- It considers success and continues 😞 

- A workaround seems possible with `--msi-url URL`
See https://docs.chef.io/workstation/knife_bootstrap/#chef-installation-options

[it appears this crafted url is working: https://packages.chef.io/files/stable/chef/15.14.0/windows/2016/chef-client-15.14.0-1-x64.msi ]

**Comment by tduffield** [comment]
So the cause of the download issues was that a chef.io redirect was missed when the website was cutover the weekend of April 17th. 

The _real_ issue is that we're using chef.io URLs in the first place. 

https://github.com/chef/chef/blob/672fd930d5eafb1c38b02eabf396655b05fb0135/knife/lib/chef/knife/core/windows_bootstrap_context.rb#L301

We _should_ be using omnitruck.chef.io endpoints.

**Comment by lamont-granquist** [comment]
This should be fixed by the redirects now and #11400 and #11401 address the url issue going forwards.
